keyboards/alt67/matrix.c

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2018 Massdrop Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "alt67.h"
+
+#include "d51_util.h"
+#include "debug.h"
+#include "clks.h"
+#include <string.h>
+
+matrix_row_t mlatest[MATRIX_ROWS];
+matrix_row_t mlast[MATRIX_ROWS];
+matrix_row_t mdebounced[MATRIX_ROWS];
+
+uint8_t row_ports[] = { MATRIX_ROW_PORTS };
+uint8_t row_pins[] = { MATRIX_ROW_PINS };
+uint8_t col_ports[] = { MATRIX_COL_PORTS };
+uint8_t col_pins[] = { MATRIX_COL_PINS };
+uint32_t row_masks[2]; //NOTE: If more than PA PB used in the future, adjust code to accomodate
+
+void matrix_init(void)
+{
+    memset(mlatest, 0, MATRIX_ROWS * sizeof(matrix_row_t));
+    memset(mlast, 0, MATRIX_ROWS * sizeof(matrix_row_t));
+    memset(mdebounced, 0, MATRIX_ROWS * sizeof(matrix_row_t));
+
+    row_masks[PA] = 0;
+    row_masks[PB] = 0;
+
+    uint8_t row;
+    for (row = 0; row < MATRIX_ROWS; row++)
+    {
+        PORT->Group[row_ports[row]].DIRCLR.reg = 1 << row_pins[row]; //Input
+        PORT->Group[row_ports[row]].OUTCLR.reg = 1 << row_pins[row]; //Low
+        PORT->Group[row_ports[row]].PINCFG[row_pins[row]].bit.INEN = 1; //Input Enable,
+        PORT->Group[row_ports[row]].PINCFG[row_pins[row]].bit.PULLEN = 1; //Pull Enable
+        row_masks[row_ports[row]] |= 1 << row_pins[row]; //Add pin to proper row mask
+    }
+
+    uint8_t col;
+    for (col = 0; col < MATRIX_COLS; col++)
+    {
+        PORT->Group[col_ports[col]].DIRSET.reg = 1 << col_pins[col]; //Output
+        PORT->Group[col_ports[col]].OUTCLR.reg = 1 << col_pins[col]; //Low
+    }
+}
```

Comment:
Please add `matrix_init_quantum();` at the end of this function.  

Without it, it doesn't add any of the QMK handling, and doesn't allow for keyboard/user customization, either.